### ISSUE TITLE
Should role=listbox with overflow fail scrollable-region-focusable?

### ISSUE THREAD

**Issue by ZoeBijl** [dequelabs/axe-core#3312]
This currently fails but I wonder if it should.

```HTML
<div role="listbox" style="overflow-y: auto; max-height: .1em;">
  <div role="option">…</div>
</div>
```

[Open in CodePen](https://codepen.io/Moiety/pen/f26fc377dbe08cdea4204cda89725a34?editors=1000) (you can use the [debug page](https://codepen.io/Moiety/debug/f26fc377dbe08cdea4204cda89725a34) to run axe)

For `role=listbox` you [should manage focus](https://www.w3.org/TR/wai-aria-1.1/#listbox) (through `aria-activedescendat`). Thus it seems to make no sense to require the region to be focusable, as it shouldn’t. Perhaps I’m unaware of situations where a `role=listbox` _should_ get focus. In which case I’d be happy to learn about those.

---

Product: axe-core
Expectation: no issue
Actual: scrollable-region-focusable fails
Motivation: not an accessibility issue

<pre><code>axe-core version: 4.3.5(?) (Whichever ember-a11y-testing uses)</code></pre>

**Comment by scottaohara** [comment]
Yes, people can manage focus via `aria-activedescendant` but _something_ needs to be focusable to do that.   Your example should fail because there's nothing focusable and the listbox is visually hidden, but still exposed (so this would still be weird if used in tandem with a focusable combobox).

For instance, [apg scrollable listbox](https://www.w3.org/TR/wai-aria-practices-1.2/examples/listbox/listbox-scrollable.html) where the listbox receives focus and then uses active descendant to manage what option is focused.

**Comment by WilcoFiers** [comment]
It's a little difficult to tell. That example is so bare-bones, I'm not too sure what is supposed to go on there. Axe-core does have an exception for listboxes that are part of a combobox, but not for a listbox by itself. I would like to see a more complete example. If it's not something you can share here, you're welcome to send it to me directly.

**Comment by ZoeBijl** [comment]
> exception for listboxes that are part of a combobox

How does it tell? A more complete example would be:

```
<label>Spec Editors
    <input type="text" role="combobox" aria-controls="combobox-list" aria-expanded="true">
</label>
<div role="listbox" aria-label="Spec Editors" id="combobox-list" style="overflow-y: auto; max-height: .1em;">
    <div role="option">…</div>
</div>
```

Which then passes in axe… I… I’ll go back and check what’s happening in our code and why it fails 🙃.

**Comment by WilcoFiers** [comment]
Here's the code for it: https://github.com/dequelabs/axe-core/blob/develop/lib/rules/scrollable-region-focusable-matches.js#L31

Axe checks if the element has a role that can be a popup, like listbox, and then it checks if that element is controlled or owned by a combobox.

**Comment by ZoeBijl** [comment]
Thank you all for looking into this. I think we can close this as it seems to be an issue in our implementation rather than the test. Thanks again!
